File: project_euler/Problem007/Problem007.java
package project_euler;

public class Problem007 {
    public void solution() {
        int target = 10001;
        int result = 1;
        int count = 1;
        while (count < target) {
            result += 2;
            if (isPrime(result)) {
                ++count;
            }
        }
        System.out.println(result);
    }

    private boolean isPrime(int value) {
        if (1 != value % 6 && 5 != value % 6) {
            return 2 == value || 3 == value;
        }

        int border = (int) Math.sqrt(value);
        for (int i = 5; i <= border; i += 6) {
            if (0 == value % i || 0 == value % (i + 2)) {
                return false;
            }
        }
        return 1 != value;
    }
}

File: project_euler/Problem007/Problem007.py
import math

class Problem007:
    def solution(self) -> None:
        target, result, count = 10001, 1, 1
        while count < target:
            result += 2
            if self.__is_prime(result):
                count += 1
        print(result)

    def __is_prime(self, value: int) -> bool:
        if 1 != value % 6 and 5 != value % 6:
            return 2 == value or 3 == value

        border = int(math.sqrt(value))
        for i in range(5, border + 1, 6):
            if not value % i or not value % (i + 2):
                return False
        return 1 != value
